src/libsync/ocsprofileconnector.cpp

```diff
@@ -0,0 +1,148 @@
+#include "ocsprofileconnector.h"
+#include "common/result.h"
+#include "networkjobs.h"
+#include "iconjob.h"
+#include "theme.h"
+
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QJsonArray>
+#include <QLoggingCategory>
+#include <QIcon>
+#include <QPainter>
+#include <QImage>
+#include <QSvgRenderer>
+#include <qpixmap.h>
+
+namespace {
+Q_LOGGING_CATEGORY(lcOcsProfileConnector, "nextcloud.gui.ocsprofileconnector", QtInfoMsg)
+
+OCC::HovercardAction jsonToAction(const QJsonObject &jsonActionObject)
+{
+    return {jsonActionObject.value("title").toString("No title"), jsonActionObject.value("icon").toString("no-icon"),
+        jsonActionObject.value("hyperlink").toString("no-link")};
+}
+
+OCC::Hovercard jsonToHovercard(const QJsonArray &jsonDataArray)
+{
+    OCC::Hovercard hovercard;
+    hovercard._actions.reserve(jsonDataArray.size());
+    for (const auto &jsonEntry : jsonDataArray) {
+        Q_ASSERT(jsonEntry.isObject());
+        if (!jsonEntry.isObject()) {
+            continue;
+        }
+        hovercard._actions.push_back(jsonToAction(jsonEntry.toObject()));
+    }
+    return hovercard;
+}
+
+OCC::Optional<QPixmap> createPixmapFromSvgData(const QByteArray &iconData)
+{
+    QSvgRenderer svgRenderer;
+    if (svgRenderer.load(iconData)) {
+        QSize imageSize{16, 16};
+        if (OCC::Theme::isHidpi()) {
+            imageSize = QSize{32, 32};
+        }
+        QImage scaledSvg(imageSize, QImage::Format_ARGB32);
+        scaledSvg.fill("transparent");
+        QPainter svgPainter{&scaledSvg};
+        svgRenderer.render(&svgPainter);
+        return QPixmap::fromImage(scaledSvg);
```

Comment:
@FlexW Is there a chance to utilize `<QPixmapCache>` to avoid generating a pixmap every time?

Reply:
Yes, why not. Need just to make sure that icons are cached per server. Not sure if they can be branded.

Reply:
@FlexW The key point is to have a unique key for a cached pixmap :) Host name could be a part of a key making it unique and per-server :) Not sure if that's what you need.

Reply:
I discovered that the URL is enough. The host name is in the URL :D so all good.